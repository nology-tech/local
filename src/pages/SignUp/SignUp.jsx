import Layout from "../../components/Layout/Layout";
const SignUp = () => {
  return (
    <Layout>
      <p>Sign Up</p>
    </Layout>
  );
};

export default SignUp;
