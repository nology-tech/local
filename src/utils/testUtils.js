import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
