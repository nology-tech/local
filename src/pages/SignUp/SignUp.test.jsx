import { customRender } from "../../utils/testUtils";
import SignUp from "./SignUp";

it("should render the text on the display", () => {
  const { container } = customRender(<SignUp />);

  expect(container).toMatchSnapshot();
});
