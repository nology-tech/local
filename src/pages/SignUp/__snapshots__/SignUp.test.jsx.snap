// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text on the display 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="sign-up"
    >
      <div
        class="sign-up__intro"
      >
        <img
          alt="App Logo"
          class="sign-up__logo"
          src="AppLogo.svg"
        />
        <h1>
          LOCAL
        </h1>
        <p>
          Bringing local business to local people
        </p>
      </div>
      <div
        class="sign-up__input-field"
      >
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            First name
          </label>
          <input
            class="text-field__input text-field__input--signup"
            type="text"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Last name
          </label>
          <input
            class="text-field__input text-field__input--signup"
            type="text"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Username
          </label>
          <input
            class="text-field__input text-field__input--signup"
            type="text"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Email
          </label>
          <input
            class="text-field__input text-field__input--signup"
            type="text"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Confirm email
          </label>
          <input
            class="text-field__input text-field__input--signup"
            type="text"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Password
          </label>
          <input
            class="text-field__input text-field__input--signup"
            type="text"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Retype password
          </label>
          <input
            class="text-field__input text-field__input--signup"
            type="text"
          />
        </div>
        <div
          class="sign-up__buttons"
        >
          <button
            class="button button--signup"
          >
            Sign up
            <img
              class="button--image--undefined"
            />
          </button>
          <button
            class="button button--signup"
          >
            Sign up with Google
            <img
              class="button--image--undefined"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
